execute/plugin.go

```diff
@@ -623,7 +623,11 @@ func (p *Plugin) checkAlreadyExecuted(
 	// TODO: batch these queries? these are all DB reads.
 	// maybe some alternative queries exist.
 	for sourceChainSelector, seqNrRange := range seqNrRangesBySource {
-		executed, err := p.ccipReader.ExecutedMessages(ctx, sourceChainSelector, seqNrRange, primitives.Unconfirmed)
+		executed, err := p.ccipReader.ExecutedMessages(
+			ctx,
+			sourceChainSelector,
+			[]cciptypes.SeqNumRange{seqNrRange},
+			primitives.Unconfirmed)
```

Comment:
With the suggested API this whole loop can be collapsed to a single call. We can change the type of `seqNrRangesBySource` to make the map value `[]SeqNumRange` instead of a single range:

```golang
executed, err := p.ccipReader.ExecutedMessages(ctx, seqNrRangesbySource, primitives.Unconfirmed)
if err != nil { ... }

// process results
```